File: simulation/backend/models/warehouse/agv.py
"""AGV navigation grid models."""
from __future__ import annotations

from typing import Literal
from pydantic import BaseModel


CellType = Literal[0, 1, 2, 3]


class AGVCell(BaseModel):
    t: CellType
    w: float


class AGVNode(BaseModel):
    x: int
    z: int
    g: float = 0
    h: float = 0
    f: float = 0
    parent: AGVNode | None = None


class AGVGrid(BaseModel):
    cols: int
    rows: int
    cell_size: float = 1.0
    cells: list[AGVCell] = []


class AisleGap(BaseModel):
    index: int
    label: str
    z_center: float
    width: float


class AGVPath(BaseModel):
    task_id: str
    points: list[tuple[float, float, float]]
    vehicle_ref: str
    status: Literal["planned", "active", "completed"] = "planned"


class AGVTool(BaseModel):
    type: Literal["block", "walk", "main", "restricted"]
    label: str
    color: str


AGV_WEIGHTS = {0: 999, 1: 1, 2: 1, 3: 5}

File: simulation/frontend/src/warehouse/types/agv.ts
export type CellType = 0 | 1 | 2 | 3

export interface AGVCell {
  t: CellType
  w: number
}

export interface AGVNode {
  x: number
  z: number
  g: number
  h: number
  f: number
  parent?: AGVNode
}

export interface AGVGrid {
  cols: number
  rows: number
  cell_size: number
  cells: AGVCell[]
}

export interface AisleGap {
  index: number
  label: string
  z_center: number
  width: number
}

export interface AGVPath {
  task_id: string
  points: [number, number, number][]
  vehicle_ref: string
  status: 'planned' | 'active' | 'completed'
}

export interface AGVTool {
  type: 'block' | 'walk' | 'main' | 'restricted'
  label: string
  color: string
}

export const AGV_TOOLS: AGVTool[] = [
  { type: 'block', label: 'Block', color: '#374151' },
  { type: 'walk', label: 'Walk', color: '#6b7280' },
  { type: 'main', label: 'Main Path', color: '#3b82f6' },
  { type: 'restricted', label: 'Restricted', color: '#f97316' },
]

export const AGV_WEIGHTS: Record<CellType, number> = {
  0: 999,
  1: 1,
  2: 1,
  3: 5,
}
